### DISCUSSION TITLE
IOM Plugin slow

### DISCUSSION THREAD

**Discussion by Rockambole** [idea-statica/ideastatica-public#442] [General Discussion]
Hello Nathan,

Just curious: even for “small” joints, running the plugin to produce my IdeaCon file takes “very” long (10-15 secs).
How is processing done? On your server or my pc (client)?

Any compiled code would – for a problem this small – not take more than a few milliseconds (assuming not too much heavy calculations are going on at this stage:)

Is this by design or am I doing something wrong?

Regards, O.S.

**Comment by nathanlukeIDEA** [comment]
Hi @Rockambole,

You will typically notice a delay 10-15 seconds on creation of the client which is basically checking license and initializing the hidden connection client. The client is running locally. 

```csharp
var CalcFactory = new ConnHiddenClientFactory(C:\Program Files\IDEA StatiCa\StatiCa 22.0);
var IdeaConnectionClient = CalcFactory.Create();
```
The actual client calls create, open, save etc should be very quick. 

Make sure you **should not** close the client until you really have to (`IdeaConnectionClient.Close()`) because then you will have to reopen it which takes time. In our projects we typically have a separate class managing the client but I am not sure how your project is working.

Hope this helps.

Cheers
Nathan

**Comment by santhoshRachamalla7** [comment]
Hi
I downloaded this code and built using VS 2017. I have installed IdeaStatica.Plugin 24.1.0.3003 and IdeaStatica.BimApiLink 24.1.0.3003 nuget packages. I am not able to install other nuget packages like IdeaStatica.OpenModel 24.1.3003, IdeaStatica.RcsClient etc. 
![BuildError](https://github.com/user-attachments/assets/34918272-0cbf-4d65-b3e9-a7c3434243b9)

I am not able to build any of the projects in IdeaStatica-SDK-Examples. Can you help me in building this solution/projects? Please let me know what nuget packages and .NET SDK versions are required.




Thanks
Santhosh.

**Reply by nathanlukeIDEA** [reply]
Hi, I believe this has to do with your version of visual studio. Please ensure a minimum version of 22. we are making some improvements to the example solutions, mainly for APP API's.
